plugins/lsp/language-servers/_mk-lsp.nix

```diff
@@ -51,9 +51,41 @@ in
     plugins.lsp.servers.${name} = {
       enable = lib.mkEnableOption description;
 
+      # alias to lsp.servers.${name}.package
       package =
-        lib.nixvim.mkMaybeUnpackagedOption "plugins.lsp.servers.${name}.package" pkgs name
-          package;
+        let
+          getSubOptions = opt: opt.type.getSubOptions opt.loc;
+          serverOpts = getSubOptions options.lsp.servers;
+          opt = lib.pipe serverOpts [
+            (lib.getAttr serverName)
+            getSubOptions
+            (lib.getAttr "package")
+          ];
+          pkg = config.lsp.servers.${serverName}.package;
+          self = options.plugins.lsp.servers.${name}.package;
+        in
+        if serverOpts ? ${serverName} then
+          lib.mkOption (
+            {
+              inherit (opt) type description;
+              apply =
+                v:
+                if enabled then
+                  pkg
+                else if self.isDefined then
+                  v
+                else
+                  opt.default or v;
+            }
+            // lib.optionalAttrs (opt ? example) { inherit (opt) example; }
+            // lib.optionalAttrs (opt ? defaultText) { inherit (opt) defaultText; }
+          )
```

Comment:
Not having a default here was intentional, as it simplifies the other side of the alias (below in `config`).

However, it seems to be causing issues with other parts of the codebase that assume they can get a default from package options. E.g. when we look for the LSP server's homepage URL in the docs, or in tests like `all-package-defaults`.